gpii/node_modules/flowManager/test/CaptureTests.js

```diff
@@ -0,0 +1,206 @@
+/**
+ * GPII Capture Tests
+ *
+ * Copyright 2019 Raising the Floor - International
+ *
+ * Licensed under the New BSD license. You may not use this file except in
+ * compliance with this License.
+ *
+ * You may obtain a copy of the License at
+ * https://github.com/GPII/universal/blob/master/LICENSE.txt
+ */
+
+"use strict";
+
+var fluid = require("infusion"),
+    gpii = fluid.registerNamespace("gpii"),
+    kettle = require("kettle");
+
+var jqUnit = require("node-jqunit");
+
+fluid.require("%gpii-universal");
+gpii.loadTestingSupport();
+
+fluid.registerNamespace("gpii.tests.flowManager.capture");
+
+kettle.config.createDefaults({
+    configName: "gpii.flowManager.tests.capture.fakeData.config",
+    configPath: "%gpii-universal/gpii/node_modules/flowManager/test/configs"
+});
+
+gpii.tests.flowManager.capture.checkForFakeMags = function (payload) {
+    jqUnit.assertDeepEq("FakeMag1 with multiple settings handlers", payload.fakemag1, {
+        "magnification": 2,
+        "invert": true
+    });
+
+    jqUnit.assertDeepEq("FakeMag2 with single settings handler", payload.fakemag2, {
+        "magnification": 2,
+        "invert": true
+    });
+};
+
+gpii.tests.flowManager.capture.platformReporter = function () {
+    return {
+        id: "darwin"
+    };
+};
+
+gpii.tests.flowManager.capture.checkForInstalledMags = function (payload) {
+    jqUnit.assertEquals("There should be 2 solutions installed", 2, Object.keys(payload).length);
+    jqUnit.assertEquals("Check for Fake Mag 1", payload.fakemag1.name, "Fake Magnifier 1");
+    jqUnit.assertEquals("Check for Fake Mag 2", payload.fakemag2.name, "Fake Magnifier 2 - fully featured");
+};
+
+gpii.tests.flowManager.capture.adjustSolutions = function (solutionsRegistryDataSource) {
+    var testDataFile = __dirname + "/data/capture_fakemag_settings.json";
+
+    // The solutions block is a read only data structure
+    var adjustedSolutions = fluid.copy(solutionsRegistryDataSource.fullSolutionsRegistry);
+    adjustedSolutions.darwin.fakemag2.settingsHandlers.configuration.options.filename = testDataFile;
```

Comment:
When all settings handlers consume same set of settings/values, it would be hard to detect a coding error that causes the same data being returned.

Reply:
this is a really good point. I created a second file, and added a few settings and make some subtle changes to the types.